src/BuildScriptGenerator/DefaultBuildScriptGenerator.cs

```diff
@@ -313,6 +336,34 @@ private void LogScriptIfGiven(string type, string scriptPath)
             buildProperties[ManifestFilePropertyKeys.CompressDestinationDir] =
                 _cliOptions.CompressDestinationDir.ToString().ToLower();
 
+            // Override the prebuild and postbuild commands if procfile exists
+            if (context.SourceRepo.FileExists("app.yaml"))
+            {
+                try
+                {
+                    var deserializer = new YamlDotNet.Serialization.DeserializerBuilder()
+                        .WithNamingConvention(CamelCaseNamingConvention.Instance)
```

Comment:
how about abstracting the section dealing with procfile parsing and error handling to a separate class altogether?

Reply:
I have moved the parsing of the Procfile to a different class. However, it doesn't seem worth it to move the error handling as I would need to pass the logger and writers to the different class and they exist in the context of this class here.